File: crypt/CipherCrypt.php
<?php

require_once "./security/SecurityHandler.php";

class CipherCrypt extends SecurityHandler {

    /** @var string $plaintext */
    private string $plaintext;

    /** @var int $sec */
    private int $sec;

    /** @var array $crypt_data */
    private array $crypt_data;

    public function __construct(string $plaintext, int $sec, $crypt_data = ["encrypted" => "", "decrypted" => ""]){
        $this->plaintext = $plaintext;
        $this->sec = $sec;
        $this->crypt_data = $crypt_data;
        $this->serialize_data($sec);
    }

    /**
     * @return string
     */
    public function encrypt() : string {
        $sec = $this->sec_data;
        $encrypt = "";

        for ($i = 0; $i < strlen($this->plaintext); $i++){
            $encrypt .= $this->hash_char($this->plaintext[$i], $sec["_"]);
        }
        return $this->crypt_data["encrypted"] = $encrypt . $sec["f_break"] . $sec["s_break"];
    }

    /**
     * @return string
     */
    public function decrypt() : string {
        $sec = $this->sec_data;
        $hash = explode(".", $this->crypt_data["encrypted"])[0];
        $decrypt = "";

        for ($i = 0; $i < strlen($hash); $i++){
            $decrypt .= $this->unhash_char($hash[$i], $sec["_"]);
        }
        return $this->crypt_data["decrypted"] = $decrypt;
    }
}

File: crypt/CipherCrypt.cpp
#include "HashAlgorithm.cpp"

class CipherCrypt : public HashAlgorithm {

private:
    std::string
        plaintext = "",
        encrypted = "",
        decrypted = "";
    int sec = 0;

public:
    CipherCrypt(std::string plaintext, int sec, std::string encrypted = "", std::string decrypted = ""){
        this->plaintext = plaintext;
        this->sec = sec;
        this->encrypted = encrypted;
        this->decrypted = decrypted;
    }

    int get_sec(){
        return this->sec;
    }

    std::string encrypt(){
        return this->hash_char("9", 5);
    }

    std::string decrypt(){
        //TODO
        return "";
    }
};
